channelserver/src/main.rs

```diff
@@ -0,0 +1,268 @@
+    &preq.state().log.do_send(logging::LogMessage {
+        level: logging::ErrorLevel::Info,
+        msg: format!("Creating session for channel: \"{}\"", channel.simple()),
+    });
+    ws::start(
+        req,
+        session::WsChannelSession {
+            id: 0,
+            hb: Instant::now(),
+            channel: channel.clone(),
+            name: None,
+        },
+    )
+}
+
+fn heartbeat(req: &HttpRequest<session::WsChannelSessionState>) -> Result<HttpResponse, Error> {
+    // if there's more to check, add it here.
+    let body = json!({"status": "ok", "version": env!("CARGO_PKG_VERSION")});
+    Ok(HttpResponse::Ok()
+        .content_type("application/json")
+        .body(body.to_string()))
+}
+
+fn lbheartbeat(req: &HttpRequest<session::WsChannelSessionState>) -> Result<HttpResponse, Error> {
+    // load balance heartbeat. Doesn't matter what's returned, aside from a 200
+    Ok(HttpResponse::Ok()
+        .content_type("application/json")
+        .body("{}"))
+}
+
+fn show_version(req: &HttpRequest<session::WsChannelSessionState>) -> Result<HttpResponse, Error> {
+    // Return the contents of the version.json file.
+    Ok(HttpResponse::Ok()
+        .content_type("application/json")
+        .body(include_str!("../version.json")))
+}
+
+fn build_app(app: App<session::WsChannelSessionState>) -> App<session::WsChannelSessionState> {
+    let mut mapp = app
+            // redirect to websocket.html
+            .resource("/", |r| r.method(http::Method::GET).f(|_| {
+                HttpResponse::NotFound()
+                    .finish()
+            }))
+            // websocket to an existing channel
+            .resource("/v1/ws/{channel}", |r| r.route().f(channel_route))
+            // connecting to an empty channel creates a new one.
+            .resource("/v1/ws/", |r| r.route().f(channel_route))
+            .resource("/__version__", |r| r.method(http::Method::GET).f(show_version))
+            .resource("/__heartbeat__", |r| r.method(http::Method::GET).f(heartbeat))
+            .resource("/__lbheartbeat__", |r| r.method(http::Method::GET).f(lbheartbeat));
+    // Only add a static handler if the static directory exists.
+    if Path::new("static/").exists() {
+        mapp = mapp.handler("/static/", fs::StaticFiles::new("static/").unwrap());
+    }
+    mapp
+}
+
+fn main() {
+    let _ = env_logger::init();
+    let sys = actix::System::new("pairsona-server");
+
+    // Start chat server actor in separate thread
+    let logger = logging::MozLogger::new();
+    let settings = settings::Settings::new().unwrap();
+    let addr = format!("{}:{}", settings.hostname, settings.port);
+    /*
+    let sessions =
+        Arc::new(Mutex::new(HashMap::<usize, Recipient<server::TextMessage>>::new()));
+    let channels =
+        Arc::new(Mutex::new(HashMap::<Uuid, HashMap<usize, server::Channel>>::new()));
+    let server = Arbiter::start(|_| server::ChannelServer::init(channels, sessions));
+    */
```

Comment:
This is left over from an issue I was seeing with the test config. When running tests, the thread globals were being reset per thread instance, so that it was impossible to test two ports talking to each other because each port instance would get a new set of coordination channels. The current resolution is to use the external integration tester and have the unit tests focus more on the internal functions. I'll remove this code.